From 94c10d4fdcd4eae6c24c9eb809d244205f3dc217 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:48 -0400
Subject: [ethernet] mlx4: Expose port number through sysfs

Message-id: <6389ed8cd7adf6bba9ae2d0fe49b95d31c6ad59f.1410425016.git.avadai@redhat.com>
Patchwork-id: 93597
O-Subject: [Patch RHEL7.1 V1 052/197] net/mlx4_en: Expose port number through sysfs
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

Initialize dev_port with port number (0 based) to be accessed through
sysfs from user space.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 76a066f2a2a0268b565459c417b59724b5a3197b)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index fad4531..caa488a 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -2342,6 +2342,7 @@ int mlx4_en_init_netdev(struct mlx4_en_dev *mdev, int port,
 
  SET_NETDEV_DEV(dev, &mdev->dev->pdev->dev);
  dev->dev_id =  port - 1;
+ dev->dev_port = port - 1;
 
  /*
   * Initialize driver private data
-- 
1.7.1